nav2_waypoint_follower/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp

```diff
@@ -31,6 +31,13 @@
 #include "image_transport/image_transport.hpp"
 
 
+/**
+ * While C++17 isn't the project standard. We have to force LLVM/CLang
+ * to ignore deprecated declarations
+ */
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Only use pragma's before and after specific lines you want to ignore, do not ignore entire files.